doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -1443,6 +1447,17 @@ Binding types or coercion classes to a notation scope
    then :g:`a` of type :g:`t` in :g:`f t a` is not recognized as an argument to
    be interpreted in scope ``scope``.
 
+  .. attr:: add_inside
+     :undocumented:
+
+  .. attr:: add_outside
+
+     These :ref:`attributes <attribute>` allow to add the new binding
+     inside or outside the already declared bindings. In absence of
```

Comment:
You need a sentence or two and perhaps an example to make "inside" and "outside" clear.  What does it mean to have multiple bindings and how would they be used?  Is it a set of bindings, a list of bindings, or something else?  It's not a stack, is it?  "inside" doesn't fit either of those.  Perhaps there's a different word that's clearer.

Optional: BTW, the existing example is wordy, the function seems too complex (distracting) and not very clear.  The key parts should be highlighted with in-line comments `(* ... *)`.  There should be an explanation of the effect on the Check output and maybe it should show what happens without the `Bind Scope`.  It would also be great to briefly explain the benefit/why users will want to use this command.  Are you willing to improve this?